### ISSUE TITLE
fused-engine: lm_head (and embeddings) I8 weights silently zeroed instead of dequantized — degenerate 'always token 0' output

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#44]
### Hardware
Strix Halo (Ryzen AI Max+ 395), Radeon 8060S iGPU + XDNA2 NPU

### OS & Kernel
Ubuntu 26.04 LTS, kernel 7.0.0-27-generic

### XRT Version
N/A — bug is in host-side model loading, not the XRT/NPU dispatch path

### Binary Version
git commit `ee0ddb1a73` (branch `fix/fused-engine-tokenizer-decode-loop`, PR #42), `engine/fusion/zig-out/bin/fused-engine`

### Exact Command
```
./engine/fusion/zig-out/bin/fused-engine -p "The capital of France is" -n 20 --policy npu_only
```

### Full Output
```
Prompt tokens (5): { 785, 6722, 315, 9625, 374 }
...
warning(model_data): lm_head has I8 dtype, expected BF16; allocating zeros
...
Generating up to 20 tokens...
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

### What's wrong

This is a follow-up to #41. PR #42 correctly fixed the tokenizer stub and the
non-autoregressive decode loop — verified independently: `"The capital of France is"`
now tokenizes to the real BPE ids `{785, 6722, 315, 9625, 374}` (matches
`AutoTokenizer.from_pretrained("Qwen/Qwen3-0.6B")` exactly), and the decode loop now
correctly feeds `last_token` back each step.

But the engine still cannot produce coherent output, for a different, deeper reason:
**the lm_head weights are silently zeroed instead of dequantized.**

`model_data.zig:842-849`:
```zig
lm_head_f32 = switch (dtype) {
    .bf16 => try loadBf16TensorFromMemory(...),
    .i8_q4nx => blk: {
        log.warn("lm_head has I8 dtype, expected BF16; allocating zeros", .{});
        const zeros = try allocator.alloc(f32, numel);
        @memset(zeros, 0);
        break :blk zeros;
    },
    ...
};
```

The production model (`model.q4nx`) stores `lm_head.weight` as I8 (Q4NX-quantized), which
hits this branch every time. Instead of dequantizing, it zero-fills the whole buffer. In
`lmHead()` (`fused_execute.zig:1982+`), every logit becomes `dot(hidden, 0) = 0` for all
151936 vocab entries, softmax over an all-equal vector is uniform, and the top-k/argmax
selection deterministically ties to index 0 — token `0`, which decodes to `"!"`. That
exactly matches the observed output: `0` repeated for every one of the 20 decode steps,
independent of the (now-correct) prompt or the (now-correct) autoregressive feedback.

The identical pattern exists for token embeddings at `model_data.zig:812-819`
(`"Embeddings have I8 dtype, expected BF16; allocating zeros"`), which would cause the
same failure mode even earlier (embedding lookup itself returns zero vectors) for any
model where embeddings aren't tied to a BF16 lm_head.

This file already has a working I8 dequantizer, `dequantizeI8Block` (line 206), used
elsewhere in this same loader for per-layer weight tiles — it's just never invoked for
the embedding/lm_head tensors.

### Fix sketch
- Replace both `.i8_q4nx => { zero-fill }` branches (embeddings at :812-819, lm_head at
  :842-849) with a real dequant path using the existing `dequantizeI8Block` (adapted for
  the `[NV, H]` embedding/lm_head shape rather than the weight-tile block layout it's
  currently called with).
- Once fixed, re-run the coherence check from #41/#42 end-to-end (real prompt tokens →
  real autoregressive decode → real dequantized logits) before upgrading "NPU fused" in
  `docs/wiki/performance.md` out of "raw throughput — output not yet fully coherent".

### Related
- #41 (tokenizer stub / decode loop) — fixed by PR #42, verified working independently.
- This issue blocks actually validating end-to-end coherence; #41's fix was necessary but
  not sufficient.

**Comment by bong-water-water-bong** [comment]
Fixed in 85d907d2. Both lm_head and embeddings I8 paths now call dequantizeI8TensorFromMemory instead of zero-filling. Verified via git blame: the dequant path is live at lines 863-865 and 892-894 of model_data.zig.
